src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -315,7 +324,17 @@ impl Coordinator {
                     optimizer_config,
                 ))
             }
-            Some(copy_to_ctx) => {
+            Some(mut copy_to_ctx) => {
+                // Getting the max worker count across replicas
+                // and using that value for the number of buckets to
+                // divide the copy to output into.
+                let max_worker_count = cluster
+                    .replicas()
+                    .map(|r| r.config.location.workers())
+                    .max()
+                    .map(u64::cast_from);
+                assert!(max_worker_count.is_some());
```

Comment:
This should always be `Some` because the check to see if there are available replicas now happen before this.